### ISSUE TITLE
Type qualifier `typing.ClassVar` is not allowed in type expressions (only in annotation expressions)

### ISSUE THREAD

**Issue by dmytro-GL** [astral-sh/ty#1187]
### Summary

`ClassVar` works fine, but `typing.ClassVar` doesn't, looks like a false positive.

```
import typing

from typing import ClassVar

import peewee


class A(peewee.Model):
    Name = peewee.CharField(255)

    class Meta:
        constraints: typing.ClassVar = [
            peewee.SQL("UNIQUE (Name)"),
        ]


class B(peewee.Model):
    Name = peewee.CharField(255)

    class Meta:
        constraints: ClassVar = [
            peewee.SQL("UNIQUE (Name)"),
        ]
```

```
$ uvx ty@0.0.1a20 check x.py
WARN ty is pre-release software and not ready for production use. Expect to encounter bugs, missing features, and fatal errors.
Checking ------------------------------------------------------------ 1/1 files                                                                                                                 error[invalid-type-form]: Type qualifier `typing.ClassVar` is not allowed in type expressions (only in annotation expressions)
  --> x.py:12:22
   |
11 |     class Meta:
12 |         constraints: typing.ClassVar = [
   |                      ^^^^^^^^^^^^^^^
13 |             peewee.SQL("UNIQUE (Name)"),
14 |         ]
   |
info: rule `invalid-type-form` is enabled by default

Found 1 diagnostic
```

RUF012 (https://docs.astral.sh/ruff/rules/mutable-class-default/) seems to be happy in both cases.

### Version

0.0.1a20

**Comment by sharkdp** [comment]
Thank you for reporting this!
